keps/sig-node/24-apparmor/README.md

```diff
@@ -0,0 +1,790 @@
+This will be enforced in API validation.
+
+If a Pod with a container specifies an AppArmor profile by field/annotation,
+then the container will only apply the Pod level field/annotation if none
+are set on the container level.
+
+To raise awareness of annotation usage (in case of old automation), a warning
+mechanism will be used to highlight that support will be dropped in v1.30.
+The mechanisms being considered are audit annotations, annotations on the
+object, events, or a warning as described in [KEP
+#1693](/keps/sig-api-machinery/1693-warnings).
+
+#### PodSecurityPolicy Support
+
+PodSecurityPolicy support has been removed in Kubernetes 1.25. Due to this,
+the equivalent functionality for annotations such as the below will not be supported:
+
+ * `apparmor.security.beta.kubernetes.io/defaultProfileName`
+ * `apparmor.security.beta.kubernetes.io/allowedProfileNames`
+
+To provide this functionality, users will need to implement equivalent rules in admission control,
+by injecting or validating a `LocalHostProfile` field for Pods or individual containers.
+
+#### Pod Update
+
+The AppArmor fields on a pod are immutable, which also applies to the
+[annotation](https://github.com/kubernetes/kubernetes/blob/b46612a74224b0871a97dae819f5fb3a1763d0b9/pkg/apis/core/validation/validation.go#L177-L182).
+
+When an [Ephemeral Container](20190212-ephemeral-containers.md) is added, it
+will follow the same rules for using or overriding the pod's AppArmor profile.
+Ephemeral container's will never sync with an AppArmor annotation.
+
+#### PodTemplates
+
+PodTemplates (e.g. ReplaceSets, Deployments, StatefulSets, etc.) will be
+ignored. The field/annotation resolution will happen on template instantiation.
+
+To raise awareness of existing controllers using the AppArmor annotations that
+need to be migrated, a warning mechanism will be used to highlight that support
+will be dropped in v1.30.
+
+The mechanisms being considered are audit annotations, annotations on the
+object, events, or a warning as described in [KEP
+#1693](/keps/sig-api-machinery/1693-warnings).
+
+#### Runtime Profiles
+
+The API Server will continue to reject annotations with runtime profiles
+different than `runtime/default`, to maintain the existing behavior.
+
+Violations would lead to the error message:
+
+```
+Invalid value: "runtime/profile-name": must be a valid AppArmor profile
+```
+
+#### Kubelet Backwards compatibility
+
+The changes brought to the Kubelet by this KEP will ensure backwards
+compatibility in a similar way the changes above define it at API Server level.
+Therefore, the AppArmor profiles will be applied following the priority order:
+
+1. Container-specific field.
+2. Container-specific annotation.
+3. Pod-wide field.
+
+In case annotations and fields at either container or pod level exist, the
+kubelet will ignore the annotations and will only apply the profile defined on
+the relevant field.
```

Comment:
This seems to contradict the above priority order, which considers the container-specific annotation before the pod field?